integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomainModelInPV.java

```diff
@@ -0,0 +1,445 @@
+
+    // build the clusterview application
+    Path distDir = buildApplication(Paths.get(APP_DIR, "clusterview"),
+        null, null, "dist", domainNamespace);
+    clusterViewAppPath = Paths.get(distDir.toString(), "clusterview.war");
+    assertTrue(clusterViewAppPath.toFile().exists(), "Application archive is not available");
+
+    V1Pod webLogicPod = setupPVPod(domainNamespace);
+
+    try {
+      logger.info("Creating directory {0} in PV", "/shared/applications");
+      Exec.exec(webLogicPod, null, false, "/bin/sh", "-c", "mkdir /shared/applications");
+    } catch (IOException | ApiException | InterruptedException ex) {
+      logger.warning(ex.getMessage());
+    }
+    try {
+      logger.info("Creating directory {0} in PV", "/shared/model");
+      Exec.exec(webLogicPod, null, false, "/bin/sh", "-c", "mkdir /shared/model");
+    } catch (IOException | ApiException | InterruptedException ex) {
+      logger.warning(ex.getMessage());
+    }
+
+    try {
+      //copy the model file to PV using the temp pod - we don't have access to PVROOT in Jenkins env
+      logger.info("Copying model file {0} to pv directory {1}",
+          Paths.get(MODEL_DIR, modelFile).toString(), "/shared/model");
+      Kubernetes.copyFileToPod(domainNamespace, webLogicPod.getMetadata().getName(), null,
+          Paths.get(MODEL_DIR, modelFile), Paths.get("shared", "model", modelFile));
+    } catch (IOException | ApiException ex) {
+      logger.warning(ex.getMessage());
+    }
+    try {
+      logger.info("Copying application file {0} to pv directory {1}",
+          clusterViewAppPath.toString(), "/shared/applications");
+      Kubernetes.copyFileToPod(domainNamespace, webLogicPod.getMetadata().getName(), null,
+          clusterViewAppPath, Paths.get("shared", "applications", "clusterview.war"));
+    } catch (IOException | ApiException ex) {
+      logger.warning(ex.getMessage());
+    }
+
+  }
+
+  /**
+   * Test domain creation from model file stored in PV.
+   * 1. Create the domain custom resource using mii with no domain and specifying a PV location for modelHome
+   * 2. Verify the domain creation is successful and application is accessible.
+   */
+  @Test
+  @DisplayName("Create MII domain with model and application file from PV")
+  public void testMiiDomainWithModelAndApplicationInPV() {
+
+    // create domain custom resource and verify all the pods came up
+    Domain domain = buildDomainResource();
+    createVerifyDomain(domain);
+
+    List<String> managedServerNames = new ArrayList<String>();
+    for (int i = 1; i <= replicaCount; i++) {
+      managedServerNames.add(MANAGED_SERVER_NAME_BASE + i);
+    }
+
+    //verify admin server accessibility and the health of cluster members
+    verifyMemberHealth(adminServerPodName, managedServerNames, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT);
+
+  }
+
+  // create custom domain resource with model file in modelHome
+  private Domain buildDomainResource() {
```

Comment:
This method looks similar to CommonMiiTestUtils.createDomainResource(). Can we use that one (make it public first), and add additional things on top of the domain created by that method?

Reply:
Using the method from CommonMiiTestUtils.createDomainResource()